tools/circleci_build_type.sh

```diff
@@ -0,0 +1,25 @@
+#!/bin/bash -ef
```

Comment:
Yes, there is definitely some cleaning to do. I was planning to do that in a separate dedicated PR but I can do that in this one if you think it's preferable.

Reply:
>we don't flake8 the diff anymore?

Unless I am missing something, these scripts were used by Travis and not by CircleCI. We are not using them since we moved to Github actions. 
After looking a little bit more into this, I really think I should clean all these scripts in a separate PR rather than on this one...

Reply:
yes I agree it is best to do it in another PR. Most were used by Travis and can be removed.
the flake8 one is a useful check and I think we should run it in github actions; WDYT?

Reply:
marking as resolved and we can pick up this discussion in #2746